src/net/events.rs

```diff
@@ -1,11 +1,13 @@
-use crate::packet::Packet;
+use crate::packet::{Fragment, Packet};
 use std::net::SocketAddr;
 
 /// Events which will be pushed through the event_receiver returned by `Socket::bind`.
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub enum SocketEvent {
     /// A packet has been received from a client.
     Packet(Packet),
+    /// An fragmentation has completed and this contains the completed data.
+    Fragment(Fragment),
```

Comment:
`Fragment` contains some other data because it doesn't have `OrderingGuarantee` or `DeliveryGuarantee`. It only has the addr and payload. So it is a trade-off between creating a `Packet` with `OrderingGuatantee` being not specified or put to `None` or leaving this here. The second reason why I did this to allow the user to handle fragments differently if they want to.

Reply:
So I ask you this question. When using TCP, do you have to (or want to) distinguish between a message that was fragmented or not?

Reply:
TCP handles its own fragmentation. We shouldn't be adding anything on top of TCP, I would think...

Reply:
@fhaynes No that is not the question here, @jstnlef yea, I thought of that too, alright let's `Packet` only.